### ISSUE TITLE
iOS Binary framework is missing bitcode

### ISSUE THREAD

**Issue by Swiftocrat** [tangrams/tangram-es#2056]
iOS
 - When I use library with CocoaPods its connecting just binary framework which I can not archive instead of whole app

How can I archive app with this binary ?
<img width="1359" alt="Знімок екрана  о 12 16 38" src="https://user-images.githubusercontent.com/36230738/56724684-2825f200-6754-11e9-9d25-d31acee39564.png">

**Comment by matteblair** [comment]
Hey @Swiftocrat thanks for the report. Can you give me a little more detail about what you're trying to do? It sounds like you have an app that includes the TangramMap framework via CocoaPods and you're getting this failure when you try to use Xcode's Archive process - is that correct?

**Comment by Swiftocrat** [comment]
Yes, that's correct

**Comment by matteblair** [comment]
Ok, I can reproduce the issue you've described on my own machine. The framework _should_ include bitcode. The `enable bitcode` option is on in the build configuration and I can verify that the `-fembed-bitcode` flag is used in both the compile and linking commands. I'll keep investigating.

**Comment by Swiftocrat** [comment]
Can I found somewhere non-bitcode version of this lib ? Because I even don't see Tangram as target in Pods project and I can do nothing with its settings.

**Comment by matteblair** [comment]
You can check out the project repository and build the framework yourself using the instructions here: https://github.com/tangrams/tangram-es/tree/master/platforms/ios 
It's not quite as simple as using it from cocoapods but I think we've made it pretty easy to run builds yourself.

**Comment by matteblair** [comment]
__tl;dr: Missing bitcode is an actual problem and I know how to fix it now__

I've finally seen this problem for myself in our Swift sample app for Tangram ES https://github.com/tangrams/tangram-swift-sample. Trying to archive that app produces the same error about being "built without full bitcode". It took a while to figure out what was going on, but I've determined now that our release builds are indeed missing full bitcode support. 

The thing that made this tricky to diagnose is that there isn't a great way to check whether a framework has bitcode or not. The best/only way I've seen is to check whether the library contains segments named `__LLVM`:
```
otool -l build/ios/Release-iphoneos/TangramMap.framework/TangramMap | grep __LLVM
```

I built framework binaries for `iphoneos` and `iphonesimulator` targets and saw that both of those had __LLVM segments. Then I checked our cocoapods universal binary and saw the output was empty! We use the `lipo` tool to create the universal binaries, maybe that's the issue? Well no, turns out I just needed to give `otool` a specific arch to inspect.
```
otool -l -arch arm64 build/ios/Release-universal/TangramMap.framework/TangramMap | grep __LLVM
```

Now I could see that the universal binary had __LLVM segments too. But it was clearly not satisfying Xcode! I checked a bit around the segment markers in the binary:
```
Section
  sectname __bundle
   segname __LLVM
      addr 0x00328000
      size 0x00000001
    offset 3194880
     align 2^0 (1)
```
The segment size is .. definitely too small. The bitcode isn't actually there, it's just a placeholder of some kind. Frustrating! But now I know that we need to change some build settings and I know what change to look for.

After reading around on similar projects it sounds like `BITCODE_GENERATION_MODE=bitcode` is the Xcode variable that we're missing. I ran a new universal build with that assignment appended to the `xcodebuild` command and checked the binary again:
```
Section
  sectname __bundle
   segname __LLVM
      addr 0x0032c000
      size 0x00e839bc
    offset 3211264
     align 2^0 (1)
```
Much more reasonable size! I was then able to drop the new framework into the Swift sample app and run an Xcode Archive - now it's working!

So to fix this for downstream consumers I need to figure out the best place to inject this property into our iOS builds and then make a release so that the Cocoapods binary gets updated.

**Comment by karimnaaji** [comment]
I have a déjà vu  reading this, wasn't that type of issue fixed by https://github.com/tangrams/tangram-es/pull/1217/commits/bff940fe70c425630c5293d28be607a41be2af88 for https://github.com/tangrams/tangram-es/issues/1147. This commit might need to be partially reintroduced.

**Comment by matteblair** [comment]
Yeah, I thought this sounded kinda familiar too! Looks like we had a working bitcode configuration at some point, then it must have regressed and we never noticed because we don't check for bitcode regularly. We can re-introduce this variable like in that commit and I'd also like to add some automated check so that we don't regress on this again.
